lib/services/vote_animation_service.dart

```diff
@@ -0,0 +1,150 @@
+import 'package:flutter/material.dart';
+import 'package:pr0gramm/constants/vote_constants.dart';
+import 'package:pr0gramm/entities/enums/vote.dart';
+import 'package:pr0gramm/entities/enums/vote_button_type.dart';
+
+enum VoteAnimation {
+  initial,
+  voted,
+  focused,
+  unfocused,
+  voteFocused,
+  voteUnfocused,
+  clearFocused,
+  clearUnfocused,
+}
+
+typedef VoteItemFn = Future Function(Vote vote);
+typedef VoteStateChangedHandler = void Function(VoteAnimation voteAnimation);
+
+class VoteAnimationService {
+  final VoteItemFn voteItemHandler;
+  final Vote initialVote;
+  Vote lastVote;
+
+  VoteAnimationService({this.voteItemHandler, this.initialVote})
+      : lastVote = initialVote;
+
+  final Map<VoteButtonType, ValueNotifier<VoteAnimation>> buttonStates =
+      new Map();
+
+  Color getInitialColor(VoteButtonType type) {
```

Comment:
Can u handle the color in the view itself? Everything else in this service is completely unrelated to a view implementation.

Reply:
could rename to getInitialState and move the color finding logic to the VoteButton.

Reply:
Sounds like a plan!

Reply:
done